feat: update base content pack with spell classes and formatting improvements

cravl-dev opened the pull request:
With much help from a School of Transmutation wizard specializing in pattern manipulation that I made for a cyberpunk campaign (his name... Regex :grin:), I have updated the base content orcbrew pack to include classes for all the spells, as listed in the SRD. I also improved the formatting of many spell descriptions for readability.

A few notes:

- I have tested the updated orcbrew file, it imports successfully and appears to work great.
- There is one spell called _Spare The Dying_ that is defined in the SRD, but isn't assigned to any classes. Of course the PHB gives it to Cleric (and then Artificer also gets it in 2014 5e, Druid in 2024 5e), but not knowing where you might draw the line regarding the licensing for content distributed with the app, I went with strictly OGL SRD only and set it to false for all classes. You can change that if you so desire.
- On the note of licensing questions, one of the things keeping me from using this app more is the lack of a toggle for the expanded spell classes introduced by Tasha's, but again, does that count as licensed content? I highly double WotC would care, and [the fantastic Spellbook app our group is currently using](https://play.google.com/store/apps/details?id=dnd.jon.spellbook) does have that feature. I can easily whip up a PR to add an `expanded-spell-lists` field if you'd like to do that.

<details>
<summary>Process description</summary>

I converted Clojure > JSON > YAML, used `yq` to create and merge the spell class lists file I made, used a lot of VSCodium regex replacements to tidy up, converted back again, and re-added the namespace decorators. I've got it scripted now for ease of editing. :smile: 

</details>

cravl-dev committed:
feat: update base content pack with spell classes and minor description formatting improvements

cravl-dev committed:
fix: add all classes false to SRD-defined but not assigned "Spare The Dying" spell

cravl-dev committed:
feat: further formatting of spell description tables and split sentences

feko006 commented:
Hey @cravl-dev ! Thanks for the interest and contribution!

Regarding the notes, the idea is to only have the openly available content as the base content of the app. The rest of the data stored is the responsibility of the person using the app so in point 2 you made the right choice! :)

As for the expanded spell list. I am personally not familiar with this, so I will take some time to familiarize myself with it first and then get back to this PR with an outlook for this.

For the time being, I have the following comments for the PR:
- The difference between the current version and the one you updated are hard to check and review due to the formatting changes. I would like to avoid manually checking each entry, so it would be nice if you can align the formatting to make this easier. E.g. provide the `androidApp/src/main/res/raw/base_content_ob.orcbrew~` using your tool, just without any edits. That should make things easier for me to check.
- The `androidApp/src/main/res/raw/base_content_ob.orcbrew~` file should be removed.

cravl-dev committed:
chore: exclude backup files

cravl-dev committed:
chore: add vscode artifacts to gitignore

cravl-dev committed:
fix: re-add top-level pack name to base content orcbrew

cravl-dev committed:
chore: autoformat orcbrew to minimize diff hits

cravl-dev commented:
@feko006 Thanks for the feedback!

I removed that backup file from the index, added such backup files and some other IDE artifacts to the `.gitignore`, and added a [`pre-update`](https://github.com/cravl-dev/generictabletoprpg/tree/pre-update) branch that you can easily diff with the `base-content-updates` branch. For some reason GitHub's compare feature is refusing to format [the diff](https://github.com/cravl-dev/generictabletoprpg/compare/cravl-dev:generictabletoprpg:pre-update...base-content-updates) correctly, at least for me. YMMV. It diffs great on desktop though, I used this:
```
git difftool -y -t meld pre-update:androidApp/src/main/res/raw/base_content_ob.orcbrew base-content-updates:androidApp/src/main/res/raw/base_content_ob.orcbrew
```

feko006 commented:
@cravl-dev awesome! I needed to set up Meld locally, but the command worked like a charm. There's a bit more to do in the code, but I'll take care of that for the next release :)

Thanks a lot for taking the time to do this contribution! The changes are quite valuable! I'll merge the PR as soon as the builds passes.

As for the expanded spell list. For now, I'll stick to the fully open and available content. But I'm open for discussing ways that it can be supported with additional content. Drop by if you want to chat about this sometime https://discord.gg/3BPR8dJU :)
